query/query.go

```diff
@@ -31,27 +31,37 @@ type Q interface {
 	String() string
 }
 
+type SearchScope uint8
+
+const (
+	ScopeNone SearchScope = iota
+	ScopeFileName
+	ScopeFileContent
+	ScopeSymbol
+)
+
 // RegexpQuery is a query looking for regular expressions matches.
 type Regexp struct {
 	Regexp        *syntax.Regexp
-	FileName      bool
-	Content       bool
+	Scope         SearchScope
 	CaseSensitive bool
 }
 
 // Symbol finds a string that is a symbol.
 type Symbol struct {
-	Atom *Substring
+	Expr Q
```

Comment:
woah! So this sort of changes the results returned by something. Did you see how I implemented `type:file` and `type:repo`? It was done similarly to `case:` where we adjust the query when we come across it. Is that possibly more appropriate?